include/fs/fs.h

```diff
@@ -170,20 +185,32 @@ struct fs_file_system_t {
 #define FS_SEEK_END	2	/* Seek from end of file.  */
 #endif
 
-
 /**
  * @brief File open
  *
- * Opens an existing file or create a new one and associates
- * a stream with it.
+ * Opens or crates, if does not exist file, depending on mode provided
+ * and associates a stream with it.
  *
  * @param zfp Pointer to file object
  * @param file_name The name of file to open
+ * @param flags The mode flags, defaults to FS_O_CREATE | FS_O_RDWR if not
+ *        provided
+ *
+ * @p flags can be empty, or combination of one or more of following flags:
+ *   FS_O_READ    open for read
+ *   FS_O_WRITE   open for write
+ *   FS_O_RDWR    open for read/write (<tt>FS_O_READ | FS_O_WRITE</tt>)
+ *   FS_O_CREATE  create file if it does not exist
+ *   FS_O_APPEND  open file at the end
  *
  * @retval 0 Success
- * @retval -ERRNO errno code if error
+ * @retval -EINVAL when bad file name is given
+ * @retval -NOENT when file path is not possible (bad mount point)
+ * @retval other negative error code, depending on file system back-end.
  */
-int fs_open(struct fs_file_t *zfp, const char *file_name);
+int fs_open(struct fs_file_t *zfp, const char *file_name, fs_mode_t flags);
+#define fs_open(f, ...) _fs_open(f, __VA_ARGS__, FS_O_CREATE | FS_O_RDWR)
```

Comment:
Need to add a comment here about what is going on with the #define and VA_ARGS

Reply:
thanks for the comment, but this will be removed (https://github.com/zephyrproject-rtos/zephyr/pull/26305#discussion_r454484395)